### DISCUSSION TITLE
`responsive` can make the video break out of the bounding box of the parent element vertically

### DISCUSSION THREAD

**Discussion by dyc3** [vimeo/player.js#992] [Bug reports]
### Expected Behavior

I expect the video player to remain within the bounding box of it's parent.

### Actual Behavior

When `responsive: true` is provided as an embed option, if the parent element's height is smaller than the video's height, the video iframe will break out of the bottom border of the parent.

### Steps to Reproduce

See my codepen: https://codepen.io/dyc3/pen/rNoObBN

My current workaround is to not use this option, and use `ResizeObserver` to manually adjust the iframe's width and height to fit the parent.

**Comment by jdreetz** [comment]
@dyc3 is your goal to crop the video or resize it so the height is 100% and there's extra white space on the left and right sides of the video?

**Comment by dyc3** [comment]
The latter, to resize the video so the height is 100% the parent, with space on the left and right.

**Comment by jdreetz** [comment]
@dyc3 are you able to use the `iframe` version of the embed? Seems like that would prevent the video from expanding beyond the bounds of the container: https://codepen.io/jdreetz-vimeo/pen/qBLNKVP

**Comment by dyc3** [comment]
My application uses the vimeo library exactly how I've set up the codepen. More verbosely, see here: https://github.com/dyc3/opentogethertube/blob/master/client/src/components/players/VimeoPlayer.vue

It specifically lets the vimeo library create and manage the iframe, rather than supplying one. The library renders an iframe wrapped in a div that looks like this:

```html
<div style="padding:75.95% 0 0 0;position:relative;">
    <iframe 
        src="https://player.vimeo.com/video/28941097?app_id=122963" 
        frameborder="0" 
        allow="autoplay; fullscreen; picture-in-picture" 
        style="position:absolute;top:0;left:0;width:100%;height:100%;" 
        title="A Trip to the Moon" 
        data-ready="true"
    ></iframe>
</div>
```
I've seen this trick before: it's a hacky workaround to get the child element (the `iframe`) to have a specific aspect ratio while remaining responsive to the size of the parent (the `div`). 

Is there a version that doesn't use iframes? I've found that `<video>` seems to play way nicer with css.

**Comment by jdreetz** [comment]
@dyc3 I wanted to clarify your requirement: is the size of the container for the embed going to be a fixed size (like in the Codepen) or dynamic size (like in the snippet in your previous post)?

> Is there a version that doesn't use iframes? I've found that <video> seems to play way nicer with css.

We only support iframe embeds currently with our SDK.

**Comment by dyc3** [comment]
In my application, the parent element has `width: 100%; height: 100%;`. Does that count as dynamic?

[The snippet here](https://github.com/vimeo/player.js/issues/949#issuecomment-1702930064) is the HTML that gets rendered by the vimeo-player library right now, as i have it in the codepen. It's not what my application does.

Would you mind elaborating more on what you're after? The issue seems clear to me, but maybe I'm missing some of the nuances here.

**Comment by jdreetz** [comment]
> In my application, the parent element has `width: 100%; height: 100%;`. Does that count as dynamic?
> 
> [The snippet here](https://github.com/vimeo/player.js/issues/949#issuecomment-1702930064) is the HTML that gets rendered by the vimeo-player library right now, as i have it in the codepen. It's not what my application does.
> 
> Would you mind elaborating more on what you're after? The issue seems clear to me, but maybe I'm missing some of the nuances here.

Thanks for followwing up with the additional answers @dyc3 . I was mostly trying to see if there was any better workarounds we could provide, but I think it's clear this is an issue we need to address in the library itself. I've filed a ticket with our team to look into the issue. We'll let you know when it's been prioritized. Thanks again for bringing this to our attention!

**Comment by bdougherty** [comment]
@dyc3 yeah this is a limitation of the padding hack. We'll be looking at moving to `aspect-ratio` at some point, but I can't guarantee when. You could use it without our official support though, something like this? https://codepen.io/bdougherty/pen/zYXpjLY

**Reply by dyc3** [reply]
Not exactly, because the videos themselves can have different aspect ratios. https://codepen.io/dyc3/pen/BaEJxeP
I suppose I could calculate the aspect ratio from the video metadata in js, but it would be kinda nice to have a css only solution

**Reply by bdougherty** [reply]
Until we update the responsive embed code to use `aspect-ratio`, I don't think there will be a CSS-only solution. You could query `getVideoWidth()` and `getVideoHeight()` when setting up each video and then set the CSS that way in the meantime maybe.
